lib/beaker/dsl/install_utils.rb

```diff
@@ -102,6 +102,91 @@ def find_git_repo_versions host, path, repository
       end
 
       #
+      # Installs `facter`, `hiera`, `puppet` and when needed `win32_ruby`
+      #
+      # @example
+      #   install_puppet_from_git(host,{
+      #    :version        => '3.6.1',
+      #    :facter_version => '2.0.1',
+      #    :hiera_version  => '1.3.3',
+      #   })
+      # @param [Host] host An object implementing {Beaker::Hosts}'s
+      #                    interface.
+      # @param [Hash{Symbol=>String}] opts
+      # @option opts [String] :repo_dir defaults to /opt/puppet-git-repos
+      # @option opts [String] :git_root The root url for your git repository, defaults to git://github.com/puppetlabs
+      # @option opts [String] :win_ruby_arch x86 or x64, defaults to x86
+      # @option opts [String] :version tag or branch to use from Puppet repo
+      # @option opts [String] :hiera_version tag or branch to use from Hiera repo
+      # @option opts [String] :facter_version tag or branch to use from Facter repo
+      #
+      # @api dsl
+      def install_puppet_from_git(host, opts = {})
+        if host.kind_of?(Array)
```

Comment:
If you want to cycle through the hosts use the block_on method for wrapping - it handles this for you.  Since this method handles multi-host is should end in _on (methods not ending in _on are assumed to execute on the default host).